mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulPrepareGameScreen.kt

```diff
@@ -27,8 +20,51 @@ class PrepareGameScreenStateImpl(private val user: AuthenticatedUser) : PrepareG
 @Composable
 fun StatefulPrepareGameScreen(
     user: AuthenticatedUser,
+    navigateToGame: (match: Match) -> Unit,
+    onCancelClick: () -> Unit,
     modifier: Modifier = Modifier,
 ) {
-  val state = remember(user) { PrepareGameScreenStateImpl(user) }
-  PrepareGameScreen(state, modifier)
+  val scope = rememberCoroutineScope()
+  val chessFacade = LocalChessFacade.current
+  val opponents =
+      remember(user) { user.following }.collectAsState(emptyList()).value.map { ProfileAdapter(it) }
+
+  val state =
+      remember(user, opponents, navigateToGame, onCancelClick, scope, chessFacade) {
+        SnapshotPrepareGameScreenState(
+            user, opponents, navigateToGame, onCancelClick, scope, chessFacade)
+      }
+  PrepareGameScreen(state = state, modifier = modifier, key = { it.uid })
+}
+
+/**
+ * An implementation of the [PrepareGameScreenState]
+ * @param user authenticated user
+ * @property colorChoice chosen color side
+ * @property selectedOpponent The selected opponent's [Profile], if any
+ */
+class SnapshotPrepareGameScreenState(
```

Comment:
Missing some fields and `@property` should not be used.

Reply:
Fixed in [3b6d9d6](https://github.com/epfl-SDP/android/pull/200/commits/3b6d9d609f7b69efe8b9d29e5165b1e8cd0b7a60)